core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -270,6 +273,9 @@ public void put(int i, Object value) {
         this.equalityIds = ArrayUtil.toIntArray((List<Integer>) value);
         return;
       case 15:
+        this.sortOrderId = (Integer) value;
```

Comment:
What about using `value != null ? (int) value : 0`?

That way, the order is always filled in as early as possible.

Reply:
Oh, I remember. This is for position delete files that don't have a sort order. That's too bad. It would be nice if we didn't need to write null for those. Nevermind, I guess.